lib/std/math/nextafter.zig

```diff
@@ -0,0 +1,329 @@
+const builtin = @import("builtin");
+const std = @import("../std.zig");
+const math = std.math;
+const assert = std.debug.assert;
+const expect = std.testing.expect;
+
+/// Returns the next representable value after `x` in the direction of `y`.
```

Comment:
Not sure, if representable is ideal as some float values are not representable NaN, +-inf and stepping into NaN is NaN. Also sNaN is not supported, but that is an overall LLVM issue.

Reply:
Most documentation for C/C++ `nextafter()` seem to use "next representable (floating-point) value" so I personally think it should be clear enough. We could update it to clarify edge cases:

> Returns the next representable value after `x` in the direction of `y`.
>
> Special cases:
> * If `x == y`, `y` is returned.
> * For floats, if either `x` or `y` is a NaN, a NaN is returned.
> * For floats, if `x == 0.0` and `@abs(y) > 0.0`, the smallest subnormal number with the sign of `y` is returned.